### ISSUE TITLE
Accessibility - Must use tab to navigate through buttons

### ISSUE THREAD

**Issue by canstudios-nicolaw** [cgkineo/adapt-quicknav#72]
We have used the NVDA screen reader. 

Normally with NVDA the user can press the down arrow to navigate through the content. However on QuickNav, when they reach the buttons, pressing down doesn't do anything. The user must press tab to navigate through the buttons. If they try to press the right arrow then NVDA starts to read out each individual character of the button. This is the case on Chrome, Firefox & IE11.

We would expect the user to be able to use the up/down arrows to navigate through the buttons. 

QuickNav v3.0.0 FW v4.0.1

Course config and screenshot to follow.

**Comment by canstudios-nicolaw** [comment]
Attaching course config.json and screenshot showing the quick nav on the course

![Screenshot from 2019-04-16 13:09:32](https://user-images.githubusercontent.com/25502448/56208421-02f4fd80-6049-11e9-8c30-8b2aa9cf5f47.png)

```
{
  "_defaultLanguage": "en",
  "_defaultDirection": "ltr",
  "_questionWeight": 1,
  "_generateSourcemap": false,
  "_forceRouteLocking": true,
  "_disableAnimationFor": [],
  "_id": "5cb473bf438c2404c6543ea4",
  "_courseId": "5cb473bf438c2404c6543ea3",
  "_drawer": {
    "_showEasing": "easeOutQuart",
    "_hideEasing": "easeInQuart",
    "_duration": 400
  },
  "_accessibility": {
    "_isEnabled": true,
    "_isTextProcessorEnabled": true,
    "_isSkipNavigationEnabled": true,
    "_ariaLevels": {
      "_menu": 1,
      "_menuItem": 2,
      "_page": 1,
      "_article": 2,
      "_block": 3,
      "_component": 4,
      "_componentItem": 5
    }
  },
  "screenSize": {
    "small": 519,
    "medium": 759,
    "large": 1024
  },
  "__v": 0,
  "_completionCriteria": {
    "_requireContentCompleted": true,
    "_requireAssessmentCompleted": false
  },
  "_logging": {
    "_isEnabled": true,
    "_level": "info",
    "_console": true
  },
  "_scrollingContainer": {
    "_isEnabled": false,
    "_limitToSelector": ""
  }
}
```

**Comment by canstudios-nicolaw** [comment]
We are keen to work on a fix for this when expected behaviour is agreed.

**Comment by oliverfoster** [comment]
The component config for the specific quicknav instance would be good also, if you've got it to hand?

**Comment by canstudios-nicolaw** [comment]
```
{
    "_component": "quicknav",
    "_classes": "",
    "_isOptional": false,
    "_isAvailable": true,
    "_isHidden": false,
    "_isVisible": true,
    "_ariaLevel": 0,
    "_disableAccessibilityState": false,
    "title": "Quick Nav Title",
    "displayTitle": "Quick Nav Title",
    "body": "<p>This is the narrative body text</p>",
    "_id": "5cb5aea4438c2404c6543ee1",
    "_parentId": "5cb48733438c2404c6543ebf",
    "_type": "component",
    "_layout": "full",
    "_onScreen": {
      "_isEnabled": false,
      "_classes": "",
      "_percentInviewVertical": 50
    },
    "instruction": "This is the quick nav instrution text",
    "_loopStyle": "none",
    "_buttons": {
      "_previous": {
        "_isEnabled": true,
        "_order": 0,
        "_classes": "",
        "_iconClass": "",
        "_alignIconRight": false,
        "text": "< Previous",
        "ariaLabel": "Previous Page",
        "_showTooltip": true,
        "tooltip": "{{displayTitle}}",
        "_customRouteId": ""
      },
      "_root": {
        "_isEnabled": true,
        "_order": 0,
        "_classes": "",
        "_iconClass": "",
        "_alignIconRight": false,
        "text": "Go to main menu",
        "ariaLabel": "Go to main menu",
        "_showTooltip": true,
        "tooltip": "{{displayTitle}}",
        "_customRouteId": ""
      },
      "_up": {
        "_isEnabled": true,
        "_order": 0,
        "_classes": "",
        "_iconClass": "",
        "_alignIconRight": false,
        "text": "Back to menu",
        "ariaLabel": "Back to menu",
        "_showTooltip": true,
        "tooltip": "{{displayTitle}}",
        "_customRouteId": ""
      },
      "_next": {
        "_isEnabled": true,
        "_order": 0,
        "_classes": "",
        "_iconClass": "",
        "_alignIconRight": false,
        "text": "Next >",
        "ariaLabel": "Next Page",
        "_showTooltip": true,
        "tooltip": "{{displayTitle}}",
        "_customRouteId": ""
      },
      "_sibling": {
        "_isEnabled": true,
        "_order": 0,
        "_classes": "",
        "_iconClass": "",
        "_alignIconRight": false,
        "text": "{{inc index}}",
        "ariaLabel": "Page {{inc index}}",
        "_showTooltip": true,
        "tooltip": "{{displayTitle}}",
        "_customRouteId": ""
      }
    }
  }
```

**Comment by canstudios-nicolaw** [comment]
Looking at the releases, I can see there is a 3.1.0...I'm going to try and reproduce this with that version (sorry I thought we were on the latest version already) Will add a comment when complete

**Comment by canstudios-nicolaw** [comment]
Same issue on 3.1.0

**Comment by oliverfoster** [comment]
This seems to be an issue with NVDA and aria-label attributes:
https://github.com/nvaccess/nvda/issues/5742
https://github.com/nvaccess/nvda/issues/4773

I haven't yet worked out if it's fixed in a downloadable version of NVDA.

**Comment by oliverfoster** [comment]
Which version of NVDA are you using?

**Comment by canstudios-nicolaw** [comment]
NVDA version 2019.1

**Comment by oliverfoster** [comment]
fixed with issue/72 found #74
